lisa/tools/tar.py

```diff
@@ -10,16 +14,60 @@ class Tar(Tool):
     def command(self) -> str:
         return "tar"
 
-    def extract(self, file: str, dest_dir: str) -> str:
+    def extract(
+        self,
+        file: str,
+        dest_dir: str,
+        strip_components: int = 0,
+    ) -> None:
         # create folder when it doesn't exist
+        assert_that(strip_components).described_as(
+            "--strip-components arg for tar must be int >= 0"
+        ).is_greater_than_or_equal_to(0)
         self.node.execute(f"mkdir -p {dest_dir}", shell=True)
-        result = self.run(
-            f"-xvf {file} -C {dest_dir}", shell=True, force_run=True, sudo=True
-        )
+        tar_cmd = f"-xvf {file} -C {dest_dir}"
+        if strip_components:
+            # optionally strip N top level components from a tar file
+            tar_cmd += f" --strip-components={strip_components}"
+        result = self.run(tar_cmd, shell=True, force_run=True, sudo=True)
         if result.exit_code != 0:
             raise LisaException(
                 f"Failed to extract file to {dest_dir}, {result.stderr}"
             )
-        # Get the folder name of the extracted file
-        source_name = result.stdout.split("/")[0]
-        return source_name
+
+    def get_tar_contents(self, file: str) -> List[str]:
+        # return a list of all the files and folders in a tar file
+        result = self.run(
+            f"-tf {file}",
+            shell=True,
+            force_run=True,
+            sudo=True,
+            expected_exit_code=0,
+            expected_exit_code_failure_message=(
+                f"Could not list items in tar file {file}"
+            ),
+        )
+        return result.stdout.splitlines()
+
+    def get_top_level_output_folders(self, file: str) -> List[str]:
```

Comment:
How about add a parameter to above `list` as `recursive`?

Reply:
I think tar -t does list recursively already so it should show all the file and folder contents of the tar

Reply:
I mean, it doesn't need to have another method to output top level folder only. We can use parameters to integrate them into one method. like `list(recursive: bool = True, is_folder_only: bool = False)`

Reply:
Ooooh I see. Yep that makes sense